File: HackerRank - CatAndMouse/solver.py
def catAndMouse(x, y, z):
    catA = abs(x - z)
    catB = abs(y - z)
    if catA < catB:
        return "Cat A"
    elif catA > catB:
        return "Cat B"
    else:
        return "Mouse C"

n = int(input())
x = []
y = []
z = []
for i in range(n):
    position = [x for x in map(int, input().split())]
    x.append(position[0])
    y.append(position[1])
    z.append(position[2])

for i in range(n):
    print(catAndMouse(x[i], y[i], z[i]))

File: HackerRank - CatAndMouse/solver.cpp
#include <iostream>

using namespace std;

int main()
{
    int q;
    cin >> q;
    while (q--)
    {
        int x, y, z;
        cin >> x >> y >> z;

        int a = abs(x - z);
        int b = abs(y - z);
        if (a < b)
        {
            cout << "Cat A" << endl;
        }
        else if (a > b)
        {
            cout << "Cat B" << endl;
        }
        else
        {
            cout << "Mouse C" << endl;
        }
    }
    return 0;
}